### PR TITLE
[MM-56008] Convert `./components/widgets/settings/bool_setting.tsx` to functional component

### LINKED ISSUE #25600: Convert ./components/widgets/settings/bool_setting.tsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-56008

### PR DESCRIPTION
**Pull request by Amir-Helali**
#### Summary

Converted ```./components/widgets/settings/bool_setting.tsx``` to a functional component. And, updated the ```bool_setting.test.tsx```.
#### Ticket Link

Fixes #25600

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: package-lock.json
[the remark this answers is not in the document]
Comment (Amir-Helali) [reply package-lock.json:1]: Done

File: webapp/channels/src/components/widgets/settings/bool_setting.test.tsx [hunk 1]
```diff
@@ -8,13 +8,16 @@ import BoolSetting from './bool_setting';
 
 describe('components/widgets/settings/BoolSetting', () => {
     test('render component with required props', () => {
+      const onChange = jest.fn();
```
[the remark this answers is not in the document]
Comment (Amir-Helali) [reply webapp/channels/src/components/widgets/settings/bool_setting.test.tsx:11]: Yeah, that was weird to me as well.

File: webapp/channels/src/components/widgets/settings/bool_setting.tsx [hunk 2]
```diff
@@ -1,56 +1,61 @@
...
 
-export default class BoolSetting extends React.PureComponent<Props> {
-    public static defaultProps: Partial<Props> = {
-        labelClassName: '',
-        inputClassName: '',
-    };
-
-    private handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
-        this.props.onChange(this.props.id, e.target.checked);
+const BoolSetting: React.FC<Props> = ({
```
[the remark this answers is not in the document]
Comment (Amir-Helali) [reply webapp/channels/src/components/widgets/settings/bool_setting.tsx]: Made the changes for this.

File: webapp/channels/src/components/widgets/settings/bool_setting.tsx [hunk 3]
```diff
@@ -1,56 +1,61 @@
...
+                        onChange={handleChange}
+                    />
+                    <span>{placeholder}</span>
+                </label>
+            </div>
+        </Setting>
+    );
 }
+
+export default BoolSetting;
```
[the remark this answers is not in the document]
Comment (Amir-Helali) [reply webapp/channels/src/components/widgets/settings/bool_setting.tsx]: made the changes for this

**Review by larkox — APPROVED** [review]
Great work!

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Amir can you check the failing lint errors?

**Comment by Amir-Helali** [comment]
@M-ZubairAhmed I tried running the ESLint fixer but It did not work for me :(

**Comment by M-ZubairAhmed** [comment]
@

> @M-ZubairAhmed I tried running the ESLint fixer but It did not work for me :(

you can do `npm run fix` it should many for you if not if will show you the errors and you can fix them manually

**Comment by larkox** [comment]
@Amir-Helali Are you able to look into this?

**Comment by Amir-Helali** [comment]
@larkox sorry I've been very busy the past couple of weeks finishing up my semester and moving to a new place, so I didn't have access to my computer. I should finish my move by the end of this week and I'll have my computer up and running then. I can take a look at it early next week if that is ok.

**Comment by larkox** [comment]
@Amir-Helali No problem. If you still want to work on this, feel free to take your time, and let us know when it is ready. For the time being I will remove the stale label.

**Comment by larkox** [comment]
@Amir-Helali Have you got any chance to look into this? Checking on you just because the automation put back the stale tag. If you need more time just let me know.

**Comment by Amir-Helali** [comment]
@larkox Hi Daniel, unfortunately my dad was hospitalized during the holidays and I haven't had the time to do any work. I think I'll have my hands full for a couple of more weeks at least with taking care of him, so I won't have the time to work on this.

**Comment by larkox** [comment]
@Amir-Helali I am really sorry to hear that. Do not worry. I will probably take over this PR and make it through the finish line. Take the time you need to take care of him.

**Comment by Amir-Helali** [comment]
@larkox thank you very much. Sorry I couldn't finish the whole thing.

**Comment by larkox** [comment]
@Amir-Helali Don't be sorry. Life happens. I am glad you did your best while you were able. Now focus on what is important now :smile:

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by larkox** [comment]
@yasserfaraazkhan This is mainly used on the bool settings in the interactive dialog, so running dialog command of the demo plugin and see everything looks nice there should be enough.

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by larkox** [comment]
Thank you @Amir-Helali For your work here :smile:

**Pull request merged**